# Level 2 Task 3 - Geographic Analysis

### Objective
Plot the locations of restaurants on a map using latitude 
and longitude coordinates and identify any patterns or 
clusters of restaurants in specific areas.

#### Libraries Used
- pandas - for data loading and analysis
- folium - for interactive map visualization

Step 1 → Load data and check latitude/longitude columns

Step 2 → Understand what latitude and longitude mean

Step 3 → Create a basic map using folium

Step 4 → Plot restaurant locations on the map

Step 5 → Observe patterns and clusters

In [2]:
import pandas as pd
import folium

df = pd.read_excel('Restaurant Analysis.xlsx')
df.columns = df.columns.str.replace('', '')

In [5]:
print(df[['Latitude', 'Longitude']].describe())

          Latitude    Longitude
count  9551.000000  9551.000000
mean     25.854381    64.126574
std      11.007935    41.467058
min     -41.330428  -157.948486
25%      28.478713    77.081343
50%      28.570469    77.191964
75%      28.642758    77.282006
max      55.976980   174.832089


#### Observation

All 9,551 restaurants have latitude and longitude values.
Latitude ranges from -41.33 to 55.97 and longitude from 
-157.94 to 174.83 showing restaurants are spread across 
multiple countries. However the median latitude of 28.57 
and longitude of 77.19 corresponds to New Delhi, India. 
This matches our finding from Task 2 where New Delhi had 
the highest number of restaurants with 5,473. So we can 
expect most markers to be clustered around New Delhi 
when we plot the map.

### Creating the Map

We create an interactive map and plot each restaurant 
as a marker using its latitude and longitude coordinates.

In [4]:
center_lat = df['Latitude'].mean()
center_lon = df['Longitude'].mean()

restaurant_map = folium.Map(location=[center_lat, center_lon], zoom_start=5)

print("Map created!", center_lat, center_lon)

Map created! 25.854380700074756 64.12657446168706


### Plotting Restaurant Locations

We add a marker for each restaurant on the map using 
its latitude and longitude coordinates.

In [6]:
# Loop through each restaurant and add a marker on the map
for index, row in df.iterrows():
    folium.CircleMarker(
        location=[row['Latitude'], row['Longitude']],
        radius=2,
        color='blue',
        fill=True
    ).add_to(restaurant_map)

print("All markers added!")

All markers added!


### Displaying and Saving the Map

We display the map and save it as an HTML file so it 
can be opened in any browser as an interactive map.

In [7]:
# Save map as HTML file
restaurant_map.save('restaurant_map.html')

print("Map saved as restaurant_map.html")

# Display map in notebook
restaurant_map

Map saved as restaurant_map.html


#### Observation

The map confirms our prediction from the describe() output. 
The largest and darkest cluster is visible in India, specifically 
around New Delhi. This directly connects to Task 2 where New Delhi 
had 5,473 restaurants — the highest in the dataset. The high 
concentration of restaurants in New Delhi explains why that 
region appears as the dominant cluster on the map.

Smaller clusters are visible in USA and Australia which also 
appeared in our city analysis as cities with multiple restaurant 
entries. Restaurants outside these clusters are spread as 
scattered individual points across Europe, Middle East, 
South Africa and Brazil showing very low restaurant density 
in those regions compared to India.